jax/lib/xla_bridge.py

```diff
@@ -289,6 +289,8 @@ def supported_numpy_dtypes():
 def normalize_to_xla_dtypes(val):
   """Normalize dtypes in a value."""
   if hasattr(val, '__array__') or np.isscalar(val):
+    if dtypes.result_type(val) == dtypes.float0:
+      return np.zeros((), dtype=bool)
```

Comment:
It may be that we could put this logic elsewhere, like in device_put handlers or something. Any thoughts?

Reply:
put in _ndarray_constant_handler.